File: CMPS101/Program Assignment 1/Lex.java
//-----------------------------------------------------------------------------
//  Name: Andy Chen
//  CruzID: achen81
//  Assignment Name: Programming Assignment 1
//-----------------------------------------------------------------------------

import java.util.*;
import java.io.*;

public class Lex {
    public static void main(String[] args) throws IOException{
        
        Scanner input = new Scanner(new File(args[0]));
        PrintWriter out = new PrintWriter(new FileWriter(args[1]));
        List sort = new List();
        int maxLength = 0;
        String increment;
        String temp;
        
        while(input.hasNextLine()) {
            maxLength++;
            increment = input.nextLine();
        }
        
        input = new Scanner(new File(args[0]));
        String[] list = new String[maxLength];
        int counter = 0;
        
        while(input.hasNextLine()) {
            list[counter] = input.nextLine();
            sort.append(counter);
            counter++;
        }

        sort.moveFront();
        int j = sort.get();
        //int k;
        if(maxLength > 1) {
            sort.moveNext();
        }
            while((sort.index >= 0)) {
                if(list[j].compareTo(list[sort.get()]) <= 0) {
                    j = sort.get();
                    sort.moveNext();
                }
                else if((sort.index >= 0) && list[j].compareTo(list[sort.get()]) > 0) {
                    sort.insertAfter(j);
                    sort.movePrev();
                    sort.delete();
                    sort.moveFront();
                    j = sort.get();
                    sort.moveNext();
            }
        }
        
        
        for(sort.moveFront(); sort.index() >= 0; sort.moveNext()) {
            out.println(list[sort.get()]);
            //System.out.println(list[sort.get()]);
        }
        input.close();
        out.close();
    }
}

File: CMPS101/Program Assignment 2/Lex.c
//-----------------------------------------------------------------------------
//  Name: Andy Chen
//  CruzID: achen81
//  Assignment Name: Programming Assignment 2
//-----------------------------------------------------------------------------
#include <stdio.h>
#include <stdlib.h>
#include "string.h"
#include "List.h"

int main(int argc, const char * argv[]) {
    FILE* infile = fopen(argv[1], "r");
    FILE* outfile = fopen(argv[2], "w");
    char string[1000];
    int maxLength = 0;
    
    if(argc != 3) {
        fprintf(outfile, "Usage: %s <input file> <output file>\n", argv[0]);
        exit(EXIT_FAILURE);
    }
    if(infile == NULL) {
        fprintf(outfile, "Unable to read from file %s\n", argv[1]);
        exit(EXIT_FAILURE);
    }
    if(outfile == NULL) {
        fprintf(outfile, "Unable to write to file %s\n", argv[2]);
        exit(EXIT_FAILURE);
    }
    
    while(fgets(string, 1000, infile)) {
        maxLength++;
    }
    
    fclose(infile);
    infile = fopen(argv[1], "r");
    
    char* list[maxLength];
    List sort = newList();
    int counter = 0;
    
    while(fgets(string, 1000, infile)) {
        list[counter] = string;
        append(sort, counter);
        counter++;
    }
    
    moveFront(sort);
    int j = get(sort);
    if(maxLength > 1) {
        moveNext(sort);
    }
    while(index(sort) >= 0) {
        continue;
        if(index(sort) >= 0 && strcmp(list[j], list[get(sort)]) <= 0) {
            j = get(sort);
            moveNext(sort);
            break;
        }
        else if(index(sort) >= 0 && strcmp(list[j], list[get(sort)]) > 0) {
            insertAfter(sort, j);
            movePrev(sort);
            delete(sort);
            moveFront(sort);
            j = get(sort);
            moveNext(sort);
            break;
        }
    }
    
    for(moveFront(sort); index(sort) >= 0; moveNext(sort)) {
        fprintf(outfile, "%s", list[get(sort)]);
    }
    
    int i = 0;
    for(i = 0; i < maxLength; i++) {
        free(list[i]);
    }
    freeList(&sort);
    fclose(infile);
    fclose(outfile);
    return(0);
}
